[Chore/Refactor] Typescript error: error TS2339: Property 'em-emoji' does not exist on type

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for refactoring, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Description

```
vscode ➜ /workspaces/dify/web (main) $ pnpm type-check 2>&1 | grep "em-emoji"
app/components/app/overview/apikey-info-panel/index.tsx(32,21): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/answer-icon/index.tsx(42,33): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/answer-icon/index.tsx(42,58): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/app-icon/index.tsx(103,40): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/app-icon/index.tsx(103,65): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/emoji-picker/Inner.tsx(118,19): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/emoji-picker/Inner.tsx(139,19): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
app/components/base/emoji-picker/Inner.tsx(174,38): error TS2339: Property 'em-emoji' does not exist on type 'JSX.IntrinsicElements'.
```

### Motivation

_No response_

### Additional Context

_No response_

--- a/web/app/components/base/emoji-picker/Inner.tsx
+++ b/web/app/components/base/emoji-picker/Inner.tsx
@@ -14,16 +14,6 @@ import Divider from '@/app/components/base/divider'
 import { searchEmoji } from '@/utils/emoji'
 import cn from '@/utils/classnames'
 
-declare global {
-  // eslint-disable-next-line ts/no-namespace
-  namespace JSX {
-    // eslint-disable-next-line ts/consistent-type-definitions
-    interface IntrinsicElements {
-      'em-emoji': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
-    }
-  }
-}
-
 init({ data })
 
 const backgroundColors = [

--- a/web/global.d.ts
+++ b/web/global.d.ts
@@ -1,3 +1,6 @@
+import './types/i18n'
+import './types/jsx'
+
 declare module 'lamejs';
 declare module 'lamejs/src/js/MPEGMode';
 declare module 'lamejs/src/js/Lame';
@@ -9,4 +12,4 @@ declare module '*.mdx' {
   export default MDXComponent
 }
 
-import './types/i18n'
+export {}

--- a/web/types/jsx.d.ts
+++ b/web/types/jsx.d.ts
@@ -1,0 +1,13 @@
+// TypeScript type definitions for custom JSX elements
+// Custom JSX elements for emoji-mart web components
+
+import 'react'
+
+declare module 'react' {
+  namespace JSX {
+    // eslint-disable-next-line ts/consistent-type-definitions
+    interface IntrinsicElements {
+      'em-emoji': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
+    }
+  }
+}

